### ISSUE TITLE
issue with using node-usb native module 

### ISSUE THREAD

**Issue by lyonlai** [SimulatedGREG/electron-vue#92]
##### Describe the issue / bug.
When you put usb inside app/package.json and import it inside a vue file. Then it tried to parse tons of files. By doing some reading from the internet, seems like putting externals: ['usb'] inside webpack config will solve the problem but then I met usb module's not found error. Any idea why?
On the other hand, when I tried to do the same in Electron React Redux boilerplate, it went smooth without any problem. Make me wondering how did they solve the problem and I started to compare the two projects. But given that I just started trying out Electron for not long, I couldn't have much clue about what's going on inside. 

##### How can I reproduce this problem?
do `npm install --save usb` in app folder then put `import usb from 'usb'; console.log(usb)`. Then you will find it.

##### Tell me about your development environment.
  * Node version: 6.9.2
  * NPM version: 3.10.9
  * vue-cli version: 2.6.0
  * Operating System: Mac OS X Sierra

**Comment by SimulatedGREG** [comment]
@lyonlai 

Sorry for the late reply, I've been away for the holidays.

Sorry for your trouble with `usb`. Its hasn't ever been the easiest thing to use native node modules within webpack. I'm glad that you mentioned Electron React boilerplate, there's some setup there that I haven't seen before and might be the answer to some other improvements I've been trying to make. It looks like their setup uses externals for all node_modules instead of bundling everything together like electron-vue does. For the moment, you might be able to add `usb` as an external and then update the ignore filter in `config.js` to include `node_modules` in the production environment.

**Comment by Flaw** [comment]
On a related note, I'm having a similar issue with nodegit. 
More specifically, nodegit depends on promisify-node which in turn uses dynamic requires (as in '(module) => require(module)') which breaks in webpack.
Currently I can only use nodegit in the main process, which is less than ideal given how electron-vue is built, and will be even more so once the main process is part of the webpack build.

@SimulatedGREG 
I've now added nodegit as an external as 'nodegit': 'commonjs nodegit', but I'm getting a module not found error in the dev build.  
Nodegit is included as a dependency in 'app/package.json', so I'd assume it should be found? 
The ignore filter you're referring to concerns only the production packaging if I'm understanding things correctly, yes?
Any pointers where I should look at?

**Comment by SimulatedGREG** [comment]
@Flaw 

Been working on migrating to webpack 2, and decided to treat dependencies as `externals`. I found that when in development, since files are served over `http` electron doesn't provide the proper context for `require` statements. You must manually add the path to `app/node_modules` for it to import properly. You can either wait out a few more days (webpack should be released [*very*](https://github.com/SimulatedGREG/electron-vue/issues/72#issuecomment-272248303) soon) and wait for a full release to `master` or take a look at the `dev` branch. Going to go ahead a close this, feel free to comment anything.

Specifically at...
https://github.com/SimulatedGREG/electron-vue/commit/033ab27231d5f94482ada7a9bad8aec9bba34cf3#diff-e4b797c6da6781c6f502d68ce67ddc2b
https://github.com/SimulatedGREG/electron-vue/commit/033ab27231d5f94482ada7a9bad8aec9bba34cf3#diff-c2daed2a6716bfe5579bc5ea723bd0c6
https://github.com/SimulatedGREG/electron-vue/commit/033ab27231d5f94482ada7a9bad8aec9bba34cf3#diff-f8f5519ac7b9fc97b564a2f1982f7dbdR16
https://github.com/SimulatedGREG/electron-vue/commit/033ab27231d5f94482ada7a9bad8aec9bba34cf3#diff-f8f5519ac7b9fc97b564a2f1982f7dbdR78

**Comment by Flaw** [comment]
I'll give the dev branch a spin when I have time. 
In the interim I took electron-react, nuked everything relating to React and fabricobbled Vue on top of it. Managed to get it working at least in dev mode, externals and all, but have no clue if packaging and such will work. 

Frankly I'd much rather use electron-vue, you've done a great job with this. As someone who writes embedded C for a living and drives himself crazy with the web side of things in his free time, the documentation has been a life saver.
